### ISSUE TITLE
Stored Procedure parameters not being passed properly

### ISSUE THREAD

**Issue by Jarrich** [Azure/azure-cosmos-dotnet-v3#675]
>We are continuously addressing and improving the SDK, if possible, make sure the problem persist in the [latest SDK version](https://www.nuget.org/packages/Microsoft.Azure.Cosmos).

**Describe the bug**
Stored Procedure parameters not being passed properly when using ExecuteStoredProcedureAsync

**To Reproduce**
Created a stored procedure in my container; that looks as follows

```
function bulkDeleteSproc(query) {
    var collection = getContext().getCollection();
    var collectionLink = collection.getSelfLink();
    var response = getContext().getResponse();
    var responseBody = {
        deleted: 0,
        continuation: true
    };
    if (!query) throw new Error("The query is undefined or null.");

    return responseBody;
}
```

I'm using the following code to test if the procedure gets called properly.

```
var query = $"SELECT nf._self FROM newsfeed nf";
           
            try
            {
                var response = await _client.GetDatabase("feeds").GetContainer("newsfeed").Scripts.ExecuteStoredProcedureAsync<BulkDeleteResponse>("bulk_delete", new PartitionKey("jarrich-user2"), parameters: new dynamic[] { query });
            }
            catch (Exception ex)
            {

            }

```

**Expected behavior**
The if (!query) check in the procedure passes, proving that the parameter was passed correctly. 

**Actual behavior**
CosmosException: Response status code does not indicate success: 400 Substatus: 0 Reason: (Message: {"Errors":["Encountered exception while executing Javascript. Exception = Error: The query is undefined or null.\r\nStack trace: Error: The query is undefined or null.\n   at bulkDeleteSproc (script.js:23:17)\n   at __docDbMain (script.js:84:5)\n   at Global code (script.js:1:2)"]}ActivityId: 538df3ac-60b3-47c4-8e5a-4acc6710bb79, Request URI: /apps/xxxxx/services/xxxxxx/partitions/0e9035fc-d073-4085-a864-bf324dbeb339/replicas/132096993248273169p/, RequestStats: RequestStartTime: 2019-08-13T07:02:32.4624811Z, RequestEndTime: 2019-08-13T07:02:33.1860943Z, Number of regions attempted: 1ResponseTime: 2019-08-13T07:02:33.1860943Z, StoreResult: StorePhysicalAddress: rntbd://cdb-ms-prod-southeastasia1-fd7.documents.azure.com:16786/apps/xxxxxx/services/91a9c510-8fab-4f39-a910-af62913e5264/partitions/0e9035fc-d073-4085-a864-bf324dbeb339/replicas/132096993248273169p/, LSN: 22, GlobalCommittedLsn: 22, PartitionKeyRangeId: 0, IsValid: True, StatusCode: 400, SubStatusCode: 0, RequestCharge: 1.98, ItemLSN: -1, SessionToken: -1#22, UsingLocalLSN: False, TransportException: null, ResourceType: StoredProcedure, OperationType: ExecuteJavaScriptAddressResolution - StartTime: 2019-08-13T07:02:32.7230157Z, EndTime: 2019-08-13T07:02:32.7816395Z, TargetEndpoint: https://xxxxxx.documents.azure.com//addresses/?$resolveFor=dbs%2f%2bAtaAA%3d%3d%2fcolls%2f%2bAtaAN5Ly3E%3d%2fdocs&$filter=protocol eq rntbd&$partitionKeyRangeIds=0, SDK: Windows/10.0.18362 cosmos-netstandard-sdk/3.1.3).

**Environment summary**
SDK Version: 3.1.0
OS Version (e.g. Windows, Linux, MacOSX): Windows 10

**Additional context**
Executing the same stored procedure from within Data Explorer in Azure does execute properly.

**Comment by Jarrich** [comment]
Happy to provide more context if required. I really need stored-procedures for my use-case, and preferably don't want to go back to the v2 SDK

**Comment by Jarrich** [comment]
Found the issue. I was using a CustomSerializer, which had TypeNameHandling set to All. This caused the dynamic[] parameters for the stored procedure call to be serialized improperly like this

`{"$type":"System.Object[], System.Private.CoreLib","$values":["Hello"]}`

Instead of just

`["Hello"]`

So I have removed my CustomSerializer and worked around my custom serialization needs in a different way.
